components/browser/tabstray/src/main/java/mozilla/components/browser/tabstray/TabsAdapter.kt

```diff
@@ -6,20 +6,34 @@ package mozilla.components.browser.tabstray
 
 import android.view.LayoutInflater
 import android.view.ViewGroup
-import androidx.annotation.LayoutRes
 import androidx.recyclerview.widget.RecyclerView
 import mozilla.components.concept.tabstray.Tabs
 import mozilla.components.concept.tabstray.TabsTray
 import mozilla.components.support.base.observer.Observable
 import mozilla.components.support.base.observer.ObserverRegistry
 
+/**
+ * Function responsible for creating a `TabViewHolder` in the `TabsAdapter`.
+ */
+typealias ViewHolderProvider = (ViewGroup, BrowserTabsTray) -> TabViewHolder
+
 /**
  * RecyclerView adapter implementation to display a list/grid of tabs.
+ * @param delegate TabsTray.Observer registry to allow `TabsAdapter` to conform to `Observable<TabsTray.Observer>`.
+ * @param viewHolderProvider a function that creates a `TabViewHolder`.
  */
 @Suppress("TooManyFunctions")
 class TabsAdapter(
     delegate: Observable<TabsTray.Observer> = ObserverRegistry(),
-    @LayoutRes private val layoutId: Int = R.layout.mozac_browser_tabstray_item
+    private val viewHolderProvider: ViewHolderProvider = { parent, tabsTray ->
+        DefaultTabViewHolder(
+                LayoutInflater.from(parent.context).inflate(
```

Comment:
Possibly a small nit is that we indented twice here.

Reply:
I'm okay with merging this since the CI build didn't complain.